packages/cli-tools/src/cacheManager.ts

```diff
@@ -26,10 +26,13 @@ function loadCache(name: string): Cache | undefined {
 }
 
 function saveCache(name: string, cache: Cache) {
-  fs.writeFileSync(
-    path.resolve(getCacheRootPath(), name),
-    JSON.stringify(cache, null, 2),
-  );
+  const fullPath = path.resolve(getCacheRootPath(), name);
+
+  const dir = path.dirname(fullPath);
+  if (!fs.existsSync(dir)) {
+    fs.mkdirSync(dir, {recursive: true}); // Create the directory and its parents if necessary
+  }
```

Comment:
If you add `force: true`, you can skip the check entirely:
```suggestion
  fs.mkdirSync(dir, {recursive: true, force: true}); // Create the directory and its parents if necessary
```

Reply:
I can't see `force` option in `mkdirSync`, seems like it's available only in `rm`

Reply:
Doh! 🤦‍♂️  You're right. I was mixing the two. But still, I don't think you actually need the check here if `recursive: true`.

Reply:
yeah you are right, removed!